test/functional/legacy/062_tab_pages_spec.lua

```diff
@@ -0,0 +1,272 @@
+      $tabnew
+      tabdo call append(line('$'), 'this is tab page ' . tabpagenr())
+      tabclose! 2
+      tabrewind
+    ]])
+    eq('this is tab page 1', eval("getline('$')"))
+    execute('undo')
+    execute('q')
+    execute('tablast')
+    eq('this is tab page 4', eval("getline('$')"))
+    execute('q!')
+
+    -- Test for settabvar() and gettabvar() functions. Open a new tab page and
+    -- set 3 variables to a number, string and a list. Verify that the
+    -- variables are correctly set.
+    source([[
+      tabnew
+      tabfirst
+      call settabvar(2, 'val_num', 100)
+      call settabvar(2, 'val_str', 'SetTabVar test')
+      call settabvar(2, 'val_list', ['red', 'blue', 'green'])
+    ]])
+
+    eq(100, eval('gettabvar(2, "val_num")'))
+    eq('SetTabVar test', eval('gettabvar(2, "val_str")'))
+    eq({'red', 'blue', 'green'}, eval('gettabvar(2, "val_list")'))
+    execute('tabnext 2')
+    eq(100, eval('t:val_num'))
+    eq('SetTabVar test', eval('t:val_str'))
+    eq({'red', 'blue', 'green'}, eval('t:val_list'))
+    execute('tabclose')
+
+    -- Test some drop features
+    -- Test for ":tab drop exist-file" to keep current window.
+    execute('sp test1')
+    execute('tab drop test1')
+    eq(1, eval('tabpagenr("$")'))
+    eq(2, eval('winnr("$")'))
+    eq(1, eval('winnr()'))
+    execute('close')
+    -- Test for ":tab drop new-file" to keep current window of tabpage 1.
+    execute('split')
+    execute('tab drop newfile')
+    eq(2, eval('tabpagenr("$")'))
+    eq(2, eval('tabpagewinnr(1, "$")'))
+    eq(1, eval('tabpagewinnr(1)'))
+    execute('tabclose')
+    execute('q')
+
+    -- Test for ":tab drop multi-opend-file" to keep current tabpage and
+    -- window.
+    execute('new test1')
+    execute('tabnew')
+    execute('new test1')
+    execute('tab drop test1')
+    eq(2, eval('tabpagenr()'))
+    eq(2, eval('tabpagewinnr(2, "$")'))
+    eq(1, eval('tabpagewinnr(2)'))
+    execute('tabclose')
+    execute('q')
+
+    execute('for i in range(9) | tabnew | endfor')
+    feed('1gt')
+    eq(1, eval('tabpagenr()'))
+    execute('tabmove 5')
+    eq(6, eval('tabpagenr()'))
+    execute('tabmove -2')
+    eq(4, eval('tabpagenr()'))
+    execute('tabmove +4')
+    eq(8, eval('tabpagenr()'))
+    execute('tabmove')
+    eq(10, eval('tabpagenr()'))
+    execute('tabmove -20')
+    eq(1, eval('tabpagenr()'))
+    execute('tabmove +20')
+    eq(10, eval('tabpagenr()'))
+    execute('3tabmove')
+    eq(4, eval('tabpagenr()'))
+    execute('7tabmove 5')
+    eq(6, eval('tabpagenr()'))
+    feed('Go<esc>')
+    execute('let a="No error caught."')
+    execute('try')
+    execute('tabmove foo')
+    execute('catch E474')
+    execute('let a="E474 caught."')
+    execute('endtry')
+    feed('i<C-R>=a<C-M><esc>')
+
+    -- Test autocommands.
+    source([[
+      tabonly!
+      let g:r=[]
+      command -nargs=1 -bar C :call add(g:r, '=== '.<q-args>.' ===')|<args>
+      function Test()
+          let hasau=has('autocmd')
```

Comment:
This should always evaluate to `1`, so should be removed. The conditionals below should be adjusted as well.